hashicorp/terraform-provider-tfe
# Add Git tags support

## Description

Adding support to Git Tags 

## Testing plan

Use resource like the following:

```tf
locals {
  organization = "..."
  oauth_token_id = "..."
  identifier = "user/repo
}

resource "tfe_workspace" "my-workspace" {
  name                  = "my-workspace"
  organization          = local.organization
  allow_destroy_plan    = true
  auto_apply            = true
  execution_mode        = "remote"
  speculative_enabled   = true
  file_triggers_enabled = false

  vcs_repo {
    identifier         = local.identifier
    ingress_submodules = false
    oauth_token_id = local.oauth_token_id
    tags_regex = "^\\d+.\\d+.\\d+$" # semantic versioning
  }
}
```

1. Verify the correct settings were set in TF Cloud/TFE VCS settings for the workspace that git tags semantic versioning is selected
   * Release a new tagged release in the repo specified in `local.identifier`
   * See a new run has been trigged
1. Change configuration to the following

```tf
locals {
  organization = "..." # set
  oauth_token_id = "..." # set
  identifier = "user/repo
  working_directory = "..." # set
}

resource "tfe_workspace" "my-workspace" {
  name                  = "my-workspace"
  organization          = local.organization
  allow_destroy_plan    = true
  auto_apply            = true
  execution_mode        = "remote"
  speculative_enabled   = true
  file_triggers_enabled = true
  working_directory = local.working_directory

  vcs_repo {
    identifier         = local.identifier
    ingress_submodules = false
    oauth_token_id = local.oauth_token_id
  }
}
```

1. Terraform plan/apply succeeds
1. Verify in TF Cloud/TFE in VCS settings git tags is not selected.

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [go-tfe documentation](https://pkg.go.dev/github.com/hashicorp/go-tfe#VCSRepo)
- [API documentation PR](https://github.com/hashicorp/terraform-docs-common/pull/64)
- [Related PR](https://github.com/hashicorp/terraform-provider-tfe/pull/xxxx)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See TESTS.md to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
$ TESTARGS="-run TestAccTFEWorkspace" make testacc

...
```
